libraries/adaptive-expressions/src/builtinFunctions/isDate.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @module adaptive-expressions
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ExpressionEvaluator } from '../expressionEvaluator';
+import { Expression } from '../expression';
+import { ReturnType } from '../returnType';
+import { ExpressionType } from '../expressionType';
+import { FunctionUtils } from '../functionUtils';
+import { MemoryInterface } from '../memory/memoryInterface';
+import { Options } from '../options';
+import {TimexProperty} from '@microsoft/recognizers-text-data-types-timex-expression';
+
+/**
+ * Return true if a given TimexProperty or Timex expression refers to a valid date.

```

Comment:
>TimexProperty or Timex expression [](start = 26, length = 33)

What is the difference?

Reply:
fixed. Timex expression -> timex string

Reply:
I'm still fuzzy on the difference.  What is a TimexProperty?  Is it a JSON structure vs. some string encoding?  If I read this documentation I would have no idea what to pass in for either.

---
In reply to: [454864718](https://github.com/microsoft/botbuilder-js/pull/2502#discussion_r454864718) [](ancestors = 454864718)

Reply:
TimexProperty is the object to describe the timex in recognizer library.

`isDate` support two input types, one is "TimexProperty" object, and another one is "timex" string, for example:
`isDate('2012-12-21T12:30')` -> true

const validFullDateTimex = new TimexProperty('2020-02-20');
`isDate(validFullDateTimex )` -> true